src/lucky_record/errors.cr

```diff
@@ -12,6 +14,13 @@ module LuckyRecord
     end
   end
 
+  # Raised when Lucky Record cannot find a record by a query_method like #first or #last
+  class RecordNotFoundError < LuckyRecordError
+    def initialize(@model : Symbol, @id : Symbol)
```

Comment:
You could put this overloaded method in the existing class so they don’t get split up. The comment could change to something like: “raided when a record could not be found”

Reply:
Of course, I'm not sure why I made it a separate class.

What does raided mean in this context?

Reply:
Oops! That was a typo. I meant “raises”